pyvrp/cpp/search/SwapStar.cpp

```diff
@@ -177,117 +249,11 @@ Cost SwapStar::evaluate(Route *routeU,
-                               best.V->tw,
-                               n(best.U)->twAfter);
-
-        deltaCost += costEvaluator.twPenalty(uTWS.totalTimeWarp());
-    }
-    else if (best.VAfter->position < best.U->position)
-    {
-        auto uTWS = TWS::merge(
-            data.durationMatrix(),
-            best.VAfter->twBefore,
-            best.V->tw,
-            routeU->twBetween(best.VAfter->position + 1, best.U->position - 1),
-            n(best.U)->twAfter);
-
-        deltaCost += costEvaluator.twPenalty(uTWS.totalTimeWarp());
-    }
-    else
-    {
-        auto uTWS = TWS::merge(
-            data.durationMatrix(),
-            p(best.U)->twBefore,
-            routeU->twBetween(best.U->position + 1, best.VAfter->position),
-            best.V->tw,
-            n(best.VAfter)->twAfter);
-
-        deltaCost += costEvaluator.twPenalty(uTWS.totalTimeWarp());
-    }
-
-    if (best.UAfter->position + 1 == best.V->position)
-    {
-        // Special case
-        auto vTWS = TWS::merge(data.durationMatrix(),
-                               best.UAfter->twBefore,
-                               best.U->tw,
-                               n(best.V)->twAfter);
-
-        deltaCost += costEvaluator.twPenalty(vTWS.totalTimeWarp());
-    }
-    else if (best.UAfter->position < best.V->position)
-    {
-        auto vTWS = TWS::merge(
-            data.durationMatrix(),
-            best.UAfter->twBefore,
-            best.U->tw,
-            routeV->twBetween(best.UAfter->position + 1, best.V->position - 1),
-            n(best.V)->twAfter);
-
-        deltaCost += costEvaluator.twPenalty(vTWS.totalTimeWarp());
-    }
-    else
-    {
-        auto vTWS = TWS::merge(
-            data.durationMatrix(),
-            p(best.V)->twBefore,
-            routeV->twBetween(best.V->position + 1, best.UAfter->position),
-            best.U->tw,
-            n(best.UAfter)->twAfter);
-
-        deltaCost += costEvaluator.twPenalty(vTWS.totalTimeWarp());
-    }
-
-    deltaCost -= costEvaluator.twPenalty(routeU->timeWarp());
-    deltaCost -= costEvaluator.twPenalty(routeV->timeWarp());
-
-    auto const uDemand = data.client(best.U->client).demand;
-    auto const vDemand = data.client(best.V->client).demand;
-
-    deltaCost += costEvaluator.loadPenalty(routeU->load() - uDemand + vDemand,
-                                           routeU->capacity());
-    deltaCost -= costEvaluator.loadPenalty(routeU->load(), routeU->capacity());
-
-    deltaCost += costEvaluator.loadPenalty(routeV->load() + uDemand - vDemand,
-                                           routeV->capacity());
-    deltaCost -= costEvaluator.loadPenalty(routeV->load(), routeV->capacity());
-
-    return deltaCost;
+    auto const deltaCostU
+        = evaluateRoute(best.U, best.V, best.VAfter, costEvaluator);
+    auto const deltaCostV
+        = evaluateRoute(best.V, best.U, best.UAfter, costEvaluator);
+    return deltaCostU + deltaCostV;
```

Comment:
Bit of a nit, but I think we can simplify this further:
```suggestion
    return evaluateMove(best.U, best.V, best.VAfter, costEvaluator)
           + evaluateMove(best.V, best.U, best.UAfter, costEvaluator);
```